packages/tests/server/validators.test.ts

```diff
@@ -320,7 +321,57 @@ test('myzod', async () => {
   await close();
 });
 
-test('arktype schema - [not officially supported]', async () => {
+test('arktype v1 schema', async () => {
+  const t = initTRPC.create();
+
+  const router = t.router({
+    num: t.procedure
+      .input(arktype.type({ text: 'string' }))
+      .query(({ input }) => {
+        expectTypeOf(input).toEqualTypeOf<{ text: string }>();
+        return {
+          input,
+        };
+      }),
+  });
+
+  const { close, client } = routerToServerAndClientNew(router);
+  const res = await client.num.query({ text: '123' });
+  expect(res.input).toMatchObject({ text: '123' });
+
+  // @ts-expect-error this only accepts a `number`
+  await expect(client.num.query('13')).rejects.toMatchInlineSnapshot(`
+	[TRPCClientError: Must be an object (was string)]
+`);
+  await close();
+});
+
+test('arktype v2 schema', async () => {
+  const t = initTRPC.create();
+
+  const router = t.router({
+    num: t.procedure
+      .input(arktype2.type({ text: 'string' }))
+      .query(({ input }) => {
+        expectTypeOf(input).toEqualTypeOf<{ text: string }>();
```

Comment:
@ssalbdivad this is the assertion erroring because `text` seems to be any

Reply:
Ahh, so there is a bug specifically in `2.0.0-beta.6` that prevents ArkType from resolving correctly with `moduleResolution: "node"`. It will be fixed in the next release.

That said, that setting should probably be updated to `NodeNext` this point anyways as the version of node associated with it is long since deprecated.

Reply:
Unfortunately changing it to NodeNext or Node16 gives lots of CJS/ESM errors. And updating the package.json to type `module` means adding file extensions to all relative imports.

I might give that a go (at least for packages/tests) and see if that's as much thread-pulling as is needed - but in a separate PR. This PR will have to wait either for that other one, or a new release of ArkType v2 beta with the bug fixed.

<img width="1120" alt="image" src="https://github.com/user-attachments/assets/fcabf54a-60aa-4031-a630-d90fcfbe7a7a">

Reply:
No issues with the new version, that will be out in the next couple days anyways.

That said, ArkType 2.0 only supports ESM which could be a separate issue depending on how this package consumes it.

Reply:
Bumped the arktype2 dependency to `2.0.0-rc.0` which fixed the type problem. Seemed to not have any ESM issues without touching tsconfig, so I'm satisfied!